jupyter/analysis-automation/BasicExample.ipynb

```diff
@@ -0,0 +1,238 @@
+    "    max_values_of_group = []    \n",
+    "    min_values_of_group = []    \n",
+    "    mean_values_of_group = []    \n",
+    "\n",
+    "    for channel in group.channels:\n",
+    "        minimum = float(get_property(channel, \"minimum\") or \"NaN\")\n",
+    "        maximum = float(get_property(channel, \"maximum\") or \"NaN\")\n",
+    "        mean_values_of_group.append((minimum + maximum) / 2)\n",
+    "        max_values_of_group.append(maximum)\n",
+    "        min_values_of_group.append(minimum)\n",
+    "\n",
+    "        \n",
+    "    # Calculate statistic values from metadata \n",
+    "    abs_min = max(max_values_of_group)\n",
+    "    abs_max = min(max_values_of_group)\n",
+    "    abs_mean = sum(mean_values_of_group) / float(len(mean_values_of_group))\n",
+    "    formatted_properties.append(\"Absolute Maximum: \"+str(round(abs_max,3))+\" °C\"+\n",
+    "                                \",Absolute Minimum: \"+str(round(abs_min,3))+\" °C\"+\n",
+    "                                \",Mean Value: \"+str(round(abs_mean,3))+\" °C\") "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "papermill": {
+     "parameters": {
+      "comment_grp_1": "Checked",
+      "show_channel_of_group": 2
+     }
+    },
+    "tags": [
+     "Parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# The Analysis Automation Procedure Package parameter cell corresponds to the ANP parameter definition. \n",
+    "# (see: cell \"property inspector\" => area: Advanced Tools => cell metadata)\n",
+    "\n",
+    "comment_grp_1 = \"Checked\"\n",
+    "show_channel_of_group = 2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The  Analysis Automation Procedure Package parameter set is available from the cell below the parameter cell itself \n",
+    "\n",
+    "# Enter the parameter in the info box of the plot. \n",
+    "formatted_properties[1] +=\",Parameter: \" + str(comment_grp_1)\n",
+    "formatted_properties[0] +=\",Channel #: \" + str(show_channel_of_group)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get bulk data from DataLink\n",
+    "bulk_data = []\n",
+    "file_id = data_links[0]['fileId']\n",
+    "grp_idx = 0\n",
+    "\n",
+    "for group in file_info.groups:\n",
+    "    chn_specs = []\n",
+    "    grp_idx += 1 \n",
+    "    chn_idx = 0\n",
+    "    \n",
+    "    for channel in group.channels:\n",
+    "        channel_specifier = tdmreader.OneChannelSpecifier(file_id=file_id, group_name=group.name, channel_name=channel.name)\n",
+    "        chn_specs.append(channel_specifier)\n",
+    "\n",
+    "    xy_chns = tdmreader.ChannelSpecificationsXyChannels(y_channels=chn_specs)\n",
+    "    channel_specs = tdmreader.ChannelSpecifications(xy_channels=[xy_chns])\n",
```

Comment:
You could reduce the calls to the TDM Reader Service to a single call, by specifying multiple xy_channels, one for each group, and retrieve the results in the data array e.g.

```python
channel_specs = tdmreader.ChannelSpecifications(xy_channels=[xy_chns_group_1, xy_chns_group_2])
...
data.data[0] # 1st group data
data.data[1] # 2nd group data
```